### PR TITLE
feat: add minimal support for Debian pre-dependencies

### PR DESCRIPTION
**Pull request by jeremy-gill**
Basic support for adding 'Pre-Depends:' to Debian packages added.

### CODE REVIEW

File: nfpm.go [hunk 1]
```diff
@@ -303,6 +305,7 @@ func (i *Info) parseSemver() {
 type Overridables struct {
 	Replaces   []string       `yaml:"replaces,omitempty" json:"replaces,omitempty" jsonschema:"title=replaces directive,example=nfpm"`
 	Provides   []string       `yaml:"provides,omitempty" json:"provides,omitempty" jsonschema:"title=provides directive,example=nfpm"`
+	Predepends []string       `yaml:"predepends,omitempty" json:"predepends,omitempty" jsonschema:"title=predepends directive,example=nfpm"`
```
Comment (caarlos0) [inline nfpm.go:308]: since this is only for debs, it should probably go into the specific deb options struct. (line 385)

**Review by caarlos0 — COMMENTED** [review]
a comment with some improvement suggestions.

can you also please update the docs and add a test?

thank you for the PR :)

**Comment by jeremy-gill** [comment]
@caarlos0, I've moved the 'predepends' tag from the platform-independent area to a Debian-specific location.  I've also tweaked `deb/deb_test.go` to provide some minimal testing of the new tag.  Finally, I updated the `www/docs/configuration.md` to explain the use of the new tag.  Is this the change you were looking for?

**Comment by caarlos0** [comment]
thank you, I'll add an integration test too.

**Pull request merged**